control/__main__.py

```
@@ -0,0 +1,36 @@
+#
+#  Copyright (c) 2021 International Business Machines
+#  All rights reserved.
+#
+#  SPDX-License-Identifier: LGPL-3.0-or-later
+#
+#  Authors: anita.shekar@ibm.com, sandy.kaur@ibm.com
+#
+
+import os
+import logging
+import argparse
+from .server import GatewayServer
+from .config import GatewayConfig
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(prog="python3 -m control",
+                                     description="Manage NVMe gateways")
+    parser.add_argument(
+        "-c",
+        "--config",
+        default="ceph-nvmeof.conf",
+        type=str,
+        help="Path to config file",
+    )
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.DEBUG)
+    logger = logging.getLogger(__name__)
+    if not os.path.isfile(args.config):
+        logger.error(f"Config file {args.config} not found.")
+        raise FileNotFoundError
```

This wouldn't catch things like `ceph-nvmeof.conf` file being there but inaccessible due to lacking permissions, etc.  I suspect the reason this check was added is that `configparser.read()` silently ignores files that it can't open, by design.  Since the gateway has a single config file location, a better fix would be to change `GatewayConfig` to use `configparser.read_file()` instead of `configparser.read()` and drop this check.